openedx/core/djangoapps/programs/utils.py

```diff
@@ -105,3 +105,35 @@ def get_programs_for_dashboard(user, course_keys):
             log.exception('Unable to parse Programs API response: %r', program)
 
     return course_programs
+
+
+def get_program_for_certificates(user, credential_programs):
+    """Build a dictionary of programs, keyed by program id.
+
+    Given a user and an iterable of credential programs, add all the programs
+    data and return them in a dictionary keyed by program ids.
+
+    Arguments:
+        user (User): The user to authenticate as when requesting program
+            certificates.
+        credential_programs (list): List of program credentials representing
+            the programs in which the given user has earned certificate.
+
+    Returns:
+        dict, containing programs keyed by program ids. Empty if programs
+        cannot be retrieved.
+    """
+    certificate_programs = []
+
+    programs = get_programs(user)
+    if not programs:
+        log.debug('No programs found for the user with ID %d.', user.id)
+        return certificate_programs
+
+    for program in programs:
```

Comment:
I have handling it https://github.com/edx/edx-platform/pull/11064/files#diff-ad7d61197fa94d2095dccbb82e6ab748R43